tests/consensus_spec/gloas/test_fixture_operations.nim

```
@@ -196,6 +197,36 @@ suite baseDescription & "Deposit Request " & preset():
       OpDepositRequestDir, suiteName, "Deposit Request", "deposit_request",
       applyDepositRequest, path)
 
+suite baseDescription & "Execution Payload " & preset():
+  func makeApplyExecutionPayloadCb(path: string): auto =
+    return proc(
+        preState: var gloas.BeaconState,
+        signed_envelope: SignedExecutionPayloadEnvelope):
+        Result[void, cstring] =
+      let payloadValid = os_ops.readFile(
+          OpExecutionPayloadDir/"pyspec_tests"/path/"execution.yaml"
+        ).contains("execution_valid: true")
+
+      var cache: StateCache
+      func executePayload(_: deneb.ExecutionPayload): bool = payloadValid
+
+      process_execution_payload(
+        defaultRuntimeConfig, preState, signed_envelope, executePayload, cache)
+
+  for path in walkTests(OpExecutionPayloadDir):
+    let
+      testDir = OpExecutionPayloadDir / "pyspec_tests" / path
+      inputFile =
+        if fileExists(testDir/"signed_envelope.ssz_snappy"):
+          "signed_envelope"
+        else:
+          continue
```

It seems unlike there are test cases meant to be ignored entirely? https://github.com/ethereum/consensus-specs/tree/v1.6.0-beta.0/tests/formats/operations#condition doesn't, though, specify how to handle these.

I see several test cases, e.g. https://github.com/ethereum/consensus-spec-tests/tree/v1.6.0-beta.0/tests/minimal/gloas/operations/execution_payload/pyspec_tests/incorrect_transaction_length_32_extra_bytes lack this `signed_envelope.ssz_snappy` file, and they all appear to be intended to be invalid.

Even so, the correct approach isn't to silently skip them, but to figure out what the principled failure mode is supposed to be, otherwise, it's basically testing the tests, rather than how Nimbus will actually use these. It's better to make it as realistic a feasible given how the tests work (e.g., this one fundamentally is a test of `process_execution_payload()` and nothing outside that universe matters).

---

I've checked with @jtraglia about the tests missing `signed_envelope.ssz_snappy`. Confirmed this is a known issue and will be fixed in the next release. These should be skipped for now

---

https://github.com/ethereum/consensus-specs/issues/4545

---

https://github.com/status-im/nimbus-eth2/pull/7552/commits/fc8836d9626ecbeddb8a1b726d99e9c47bb316fc